### ISSUE TITLE
Implement a boolean/toggle setting that allows us to switch between using local and non-local dependencies.

### ISSUE THREAD

**Issue by Kietyo** [korlibs/kproject#22]
Idea: 
- Add a new dependency list called **localDependencies** to **NewKProjectModel**
- Add a boolean setting called **useLocal**
- If useLocal=true, then the project should use **localDependencies** over **dependencies**
- If useLocal=false, then the project should use **dependencies** over **localDependencies**

Use case: 
- If I post the repo on github, I would have **useLocal** set to **false**, so it should use non-local (e.g maven/github) sources so that when people fork it, everything will work out-of-the-box.
- When I'm developing locally, I would switch to using local dependencies (**useLocal=true**). E.g: So that I would source from local dirs for XenoECS and ktruth for my own project so I could develop those libraries as well while developing the main project (XenoTactic)

**Comment by jobe-m** [comment]
Makes sense. Currently when I develop I have a local copy and I need to set it up separately. E.g. I have a git submodule for korge-fleks in my project and add it manually with kproject. Not a big deal but if it would work automatically it would be easier for newbie's;)

**Comment by soywiz** [comment]
Let me brainstorm a bit:

Thinking about it, what about adding a git submodule or doing a partial git checkout that would actually allow commiting or something like that?

The problem I see with that approach is that you would have to actually remember to update the commit hash of your dependency, or the code would stop working for other people?

What about for example include your Game + the library you want to edit in your repo. Then after some changes and when you are satisfied with it, copy it into another repo, and make a release tag?

**Comment by Kietyo** [comment]
thx for the suggestion, ill try to use submodules and see what that's like

**Comment by jobe-m** [comment]
> What about for example include your Game + the library you want to edit in your repo. Then after some changes and when you are satisfied with it, copy it into another repo, and make a release tag?

I would like to avoid the need to copy over my changes. Git submodules work fine if you know how to set it up for kproject.

Maybe if we implement an automated way of copying changes from the module subfolder, where kproject places the code inside the project, back to the git repo under HOME/.kproject/..., where kproject clones the actual repo, than the PR could be made from that repo.
